src/test/scala/mesosphere/marathon/api/akkahttp/v2/EventsControllerTest.scala

```diff
@@ -0,0 +1,90 @@
+package mesosphere.marathon
+package api.akkahttp.v2
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.event.EventStream
+import akka.http.scaladsl.model.RemoteAddress
+import akka.stream.{ ActorMaterializer, ThrottleMode }
+import akka.stream.scaladsl.{ Sink, Source }
+import com.typesafe.config.{ Config, ConfigFactory }
+import mesosphere.UnitTest
+import mesosphere.marathon.api.akkahttp.v2.EventsController.EventStreamSourceGraph
+import mesosphere.marathon.core.event.{ ApiPostEvent, MarathonEvent }
+import mesosphere.marathon.state.{ AppDefinition, PathId }
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.time.{ Millis, Second, Seconds, Span }
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class EventsControllerTest extends UnitTest {
+  import EventsControllerTest._
+  implicit val system = ActorSystem("EventsControllerTest", ConfigFactory.load(testConf))
+  implicit val materializer = ActorMaterializer()
+
+  "EventStreamSourceGraph" should {
+    "be a valid source" in {
+      val fakeEventBus: EventStream = new EventStream(system)
+      val sourceUnderTest = Source
+        .fromGraph[MarathonEvent, NotUsed](
+          new EventStreamSourceGraph(fakeEventBus, 5, fakeClientIp)
+        )
+      val fut = sourceUnderTest.take(3).runWith(Sink.seq)
+      Thread.sleep(200) // avoid race with materialization of the stream
+      fakeEventBus.publish(fakeApiPostEvent)
+      Thread.sleep(200)
+      fakeEventBus.publish(fakeApiPostEvent)
+      val result = Await.result(fut, 5 seconds)
+      println(result)
+      assert(result.tail.head == fakeApiPostEvent)
+    }
+
+    "fail for queue overload" in {
+      val fakeEventBus: EventStream = new EventStream(system)
+      val sourceUnderTest = Source
+        .fromGraph[MarathonEvent, NotUsed](
+          new EventStreamSourceGraph(fakeEventBus, 1, fakeClientIp)
+        )
+      val fut = sourceUnderTest.take(3).throttle(1, 1 second, 1, ThrottleMode.shaping).runWith(Sink.seq)
+
+      Thread.sleep(200) // avoid materialization race
+      var i: Int = 0
```

Comment:
How about `(1 to 20).foreach(fakeEventBus.publish(fakeApiPostEvent))`?

Reply:
The whole test will be re-worked, see comments above.